src/main/java/ch/njol/skript/expressions/ExprBlockData.java

```diff
@@ -1,85 +1,115 @@
-/**
- *   This file is part of Skript.
- *
- *  Skript is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  Skript is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
- *
- * Copyright Peter Güttinger, SkriptLang team and contributors
- */
 package ch.njol.skript.expressions;
 
-import org.bukkit.block.Block;
-import org.bukkit.block.data.BlockData;
-import org.bukkit.event.Event;
-import org.jetbrains.annotations.Nullable;
-
-import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.BlockDataUtils;
 import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.BlockDataMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
 
 @Name("Block Data")
-@Description("Get the <a href='classes.html#blockdata'>block data</a> associated with a block. This data can also be used to set blocks.")
-@Examples({"set {data} to block data of target block",
+@Description({
+	"Get the <a href='classes.html#blockdata'>block data</a> associated with a block or itemtype.",
+	"This data can also be used to set blocks, or the set the blockdata of a valid item."
+})
+@Examples({
+	"set {data} to block data of target block",
 	"set block at player to {data}",
-	"set block data of target block to oak_stairs[facing=south;waterlogged=true]"})
+	"set block data of target block to oak_stairs[facing=south;waterlogged=true]",
+	"",
+	"set {data} to block data of oak log",
+	"set block data of {data} to oak_log[axis=z]"
+})
 @RequiredPlugins("Minecraft 1.13+")
```

Comment:
2.9 will be 13+ only

Reply:
I know, but I think its still valuable to leave it there so people have no reason to be confused if they see it on the docs and wonder why it isnt working on their older than 1.13 server